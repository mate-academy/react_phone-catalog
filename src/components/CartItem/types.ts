import { CartObject } from '../../store/models/cart';

export interface CartItemProps {
  cartItem: CartObject,
  onIncrement: (id: string) => void,
  onDecrement: (id: string) => void,
  onRemove: (id: string) => void,
}
