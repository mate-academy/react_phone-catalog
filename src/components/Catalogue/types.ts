import { Product } from '../../store/models/product';

export interface CatalogueProps {
  items: Product[];
}
