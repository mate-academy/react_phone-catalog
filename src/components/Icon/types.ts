import { Icons } from '../../types/Icons';

export interface IconProps {
  icon: Icons,
  counter?: number,
}
