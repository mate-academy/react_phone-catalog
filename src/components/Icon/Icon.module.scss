@import "../../styles/variables.scss";

.baseIcon {
  padding: 8px;
  border: 1px solid $Icons;
  background: $White;

  &:hover {
    border-color: $Primary;
  }
}

.disabled {
  opacity: 0.5;
  border-color: $Elements !important;
  cursor: not-allowed;
}
