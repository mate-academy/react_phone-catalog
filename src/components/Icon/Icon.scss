@import '../../App';

.icon {
  position: relative;
  display: flex;

  width: 16px;
  height: 16px;

  &__image {
    display: block;

    width: 16px;
    height: 16px;

    background-position: center;
    background-repeat: no-repeat;

    &-search {
      background-image: url(../../assets/icons/Search.svg);
    }

    &-cart {
      background-image: url(../../assets/icons/cart.svg);
    }

    &-heart {
      background-image: url(../../assets/icons/heart.svg);

      &-active {
        background-image: url(../../assets/icons/HeartActive.svg);
      }
    }

    &-arrowLeft {
      background-image: url(../../assets/icons/arrowLeft.svg);
    }

    &-arrowRight {
      background-image: url(../../assets/icons/arrowRight.svg);
    }

    &-arrowDown {
      background-image: url(../../assets/icons/arrowRight.svg);
      rotate: 90deg;
    }


    &-arrowUp {
      background-image: url(../../assets/icons/arrowRight.svg);
      rotate: 270deg;
    }

    &-home {
      background-image: url(../../assets/icons/Icons_Home.svg)
    }

    &-close {
      background-image: url(../../assets/icons/Close.svg)
    }
  }

  &__counter {
    @include font-small;

    position: absolute;
    top: -4px;
    right: -4px;

    border: 1px solid $white-color;
    border-radius: 50%;

    display: flex;

    justify-content: center;
    align-items: center;

    font-size: 9px;
    color: $white-color;

    width: 14px;
    height: 14px;
    background-color: $red-color;
  }
}
