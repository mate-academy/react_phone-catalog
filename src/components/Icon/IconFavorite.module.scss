@import '../../styles/main.scss';

.icon {
  display: flex;
  padding-inline: 24.3px;
  padding-block: 25.3px;

  &__img {
    width: 16px;
    height: 14px;
  }
}
