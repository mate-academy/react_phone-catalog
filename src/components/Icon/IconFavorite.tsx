import React from 'react';
import styles from './IconFavorite.module.scss';
import favouriteWhite from '../../assets/images/Icons/icon-liked-white.png';

export const IconFavorite = () => {
  return (
    <div className={styles.icon}>
      <img alt="favourites" src={favouriteWhite} className={styles.icon__img} />
    </div>
  );
};
