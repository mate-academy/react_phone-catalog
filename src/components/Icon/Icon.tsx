import { memo } from 'react';
import cn from 'classnames';
import { IconProps } from './types';
import './Icon.scss';

export const Icon = memo<IconProps>(({ icon, counter = 0 }) => (
  <div className="icon">
    <span
      className={cn('icon__image', {
        [`icon__image-${icon}`]: icon,
      })}
    />
    {counter > 0 && (
      <span className="icon__counter">
        {counter}
      </span>
    )}
  </div>
));
