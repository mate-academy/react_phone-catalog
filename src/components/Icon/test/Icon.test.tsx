import { render } from '@testing-library/react';
import { Icon } from '../Icon';
import { Icons } from '../../../types/Icons';

describe('Icon component', () => {
  it('renders with icon', () => {
    const component = render(<Icon icon={Icons.ArrowUp} />);

    expect(component).toMatchSnapshot();
  });
});
