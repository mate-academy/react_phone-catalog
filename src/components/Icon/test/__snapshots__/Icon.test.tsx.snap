// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component renders with error message 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="icon"
      >
        <span
          class="icon__image icon__image-arrowUp"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="icon"
    >
      <span
        class="icon__image icon__image-arrowUp"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Icon component renders with icon 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="icon"
      >
        <span
          class="icon__image icon__image-arrowUp"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="icon"
    >
      <span
        class="icon__image icon__image-arrowUp"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
